### PR TITLE
blog: whats new in axum 0.6.0-rc.1

### PR DESCRIPTION
**Pull request by davidpdrsn**
[🖼 Rendered](https://github.com/tokio-rs/website/blob/david/axum-0.6/content/blog/2022-08-whats-new-in-axum-0-6.md)

### CODE REVIEW

File: content/blog/2022-08-whats-new-in-axum-0-6.md [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+in a few weeks.
+
+<div style="text-align:right">&mdash; David Pedersen (<a href="https://twitter.com/davidpdrsn">@davidpdrsn</a>)</div>
+
+[state docs]: TODO
+[`FromRequest`]: TODO
+[`FromRequestParts`]: TODO
+[`middleware::from_fn`]: TODO
+[`RouterExt::route_with_tsr`]: TODO
+[extract state in middleware]: TODO
```
Comment (davidpdrsn) [inline content/blog/2022-08-whats-new-in-axum-0-6.md:273]: Note to self: Fix these when 0.6-rc.1 is actually published.

File: content/blog/2022-08-whats-new-in-axum-0-6-0-rc1.md [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+    // And extract our shared `AppState` using `Extension`
+    Extension(state): Extension<AppState>,
+) {}
+```
+
+However this wasn't type safe, so if you forgot `.layer(Extension(...))` things
+would compile just fine but you'd get runtime errors when calling `handler`.
+
+In 0.6 you can use the new `State` extractor which works similarly to
+`Extension` but is type safe:
```
Comment (LucioFranco) [inline content/blog/2022-08-whats-new-in-axum-0-6-0-rc1.md:49]: Type safe or statically checked? I don't feel like that is what type safe means right?

File: content/blog/2022-08-whats-new-in-axum-0-6-0-rc1.md [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+    // In 0.5 these routes would be considered overlapping and not be allowed
+    // but in 0.6 it just works
+    .route("/foo/*rest", get(|| async {}))
+    .route("/foo/bar", get(|| async {}));
+```
+
+## See the changelog for more
+
+I encourage you to read the [changelog] to see all the changes and for tips on
+how to update from 0.5 to 0.6.0-rc.1.
```
Comment (LucioFranco) [inline content/blog/2022-08-whats-new-in-axum-0-6-0-rc1.md:260]: Is there a timelines for `0.6`?

**Review by LucioFranco — APPROVED** [review]
LGTM just two small questions

File: content/blog/2022-08-whats-new-in-axum-0-6-0-rc1.md (hunk 2 above)
Comment (davidpdrsn) [reply content/blog/2022-08-whats-new-in-axum-0-6-0-rc1.md:49]: Uhm whats the difference? 😅

I mean that if you attempt to extract the wrong state type then it wont compile.

File: content/blog/2022-08-whats-new-in-axum-0-6-0-rc1.md (hunk 3 above)
Comment (davidpdrsn) [reply content/blog/2022-08-whats-new-in-axum-0-6-0-rc1.md:260]: Nope. I don't wanna set any expectations I might not be able to meet. But my estimate is two weeks, depending on how many issues people find.

**Pull request merged**